system-helper/flatpak-system-helper.c

```diff
@@ -791,6 +791,49 @@ handle_update_remote (FlatpakSystemHelper *object,
 }
 
 static gboolean
+handle_remove_local_ref (FlatpakSystemHelper *object,
+                         GDBusMethodInvocation *invocation,
+                         const gchar *arg_remote,
+                         const gchar *arg_ref,
+                         const gchar *arg_installation)
+{
+  g_autoptr(FlatpakDir) system = NULL;
+  g_autoptr(GError) error = NULL;
+
+  g_debug ("RemoveLocalRef %s %s %s", arg_remote, arg_ref, arg_installation);
+
+  system = dir_get_system (arg_installation, &error);
+  if (system == NULL)
+    {
+      g_dbus_method_invocation_return_gerror (invocation, error);
+      return TRUE;
+    }
+
```

Comment:
Maybe this should fail if the specified ref is already deployed at the same version? It seems weird to allow the system repo to end up in a somewhat inconsistent state.

Reply:
I'm not sure I understand what is meant here. If you are saying that we should not be permitted to remove refs from the local repo if they are already deployed, I'm not sure what impact that would have - I suppose the user would to re-pull the ref. I suppose what I could do is add a check to `flatpak_dir_remove_ref` that checks if the ref is deployed for now.

Reply:
Hmm, actually, I don't think that will work. It is possible to remove a remote and at that point all refs are removed from the local repo. This would put the repo in the same "inconsistent state".